src/components/CreateCourse/CreateCourse.tsx

```diff
@@ -0,0 +1,165 @@
+import React, { useState } from 'react';
+import Button from '../../common/Button/Button';
+import Input from '../../common/Input/Input';
+import { Author } from '../../constants';
+import { formatCreationDate } from '../../helpers/formatCreationDate';
+import { getCourseDuration } from '../../helpers/getCourseDuration';
+import AuthorItem from './components/AuthorItem/AuthorItem';
+import { addCourse } from '../../helpers/getCourseData';
+import { addAuthor } from '../../helpers/getAuthorData';
+import { getAllAuthors } from '../../helpers/getAuthorData';
+import './styles.scss';
+
+export default function CreateCourse(props: {
+	toggleCreateCourse: () => void;
+}) {
+	const [title, setTitle] = useState('');
+	const [description, setDescription] = useState('');
+	const [authorName, setAuthorName] = useState('');
+	const [duration, setDuration] = useState<number>(0);
+	const [courseAuthors, setCourseAuthors] = useState<Author[]>([]);
+	const [otherAuthors, setOtherAuthors] = useState<Author[]>(getAllAuthors());
+
+	const handleSubmit = () => {
+		if (
+			title.length < 2 &&
+			description.length < 2 &&
+			courseAuthors.length < 1 &&
+			duration <= 0
+		) {
+			alert('Invalid input! Try again!');
+		} else {
+			let authorsWithOnlyIDs = courseAuthors.map((author) => author.id);
+			const newCourse = {
+				id: generateId(),
+				title,
+				description,
+				duration,
+				authors: authorsWithOnlyIDs,
+				creationDate: formatCreationDate(new Date()),
+			};
+
+			addCourse(newCourse);
+			props.toggleCreateCourse();
+			setTitle('');
```

Comment:
It seems we may avoid using setting default values, as we will re-init component next time